### PR TITLE
[MM-51854] Onboarding Role selection screen

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
You're probably watching the number of file changes and you are panicking right now. It's gonna be ok! A lot of them are new pictures that have been added.

The main goal of this PR is to add a "Role" selection screen for the first admin based on work template categories, but as a consequence, I had to add a lot of new work template categories + content.

That role is stored in the system store (alongside the org name) for later use.

### QA

you must create a `/cloud` instance:

`/cloud create YOURNAME-pr-23121  --image mattermostdevelopment/mm-ee-test --version 062ffc2`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51854

#### Screenshots

https://user-images.githubusercontent.com/785518/234378776-24a24c04-03d9-45c7-b263-8c6c469ca03c.mp4

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/preparing_workspace/roles.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                                </button>
+                            ))}
+                        </div>
+                        {role === CategoryOther && (
+                            <Input
+                                value={props.roleOther}
+                                onChange={(e) => props.setRole(CategoryOther, e.target.value)}
+                                placeholder={formatMessage({
+                                    id: 'onboarding_wizard.roles.other_input_placeholder',
+                                    defaultMessage: 'Please share your primary function.',
```
Comment (julmondragon) [inline webapp/channels/src/components/preparing_workspace/roles.tsx:117]: nit: placeholders/labels usually don't use the period at the end.

File: webapp/channels/src/components/preparing_workspace/roles.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    const selectRole = (role: string) => {
+        if (role !== CategoryOther) {
+            props.quickNext(role);
+        }
+        props.setRole(role, '');
+    };
+
+    const roleIsSet = Boolean(role);
+    const roleOtherIsSet = Boolean(props.roleOther) && props.roleOther !== '';
```
Comment (julmondragon) [inline webapp/channels/src/components/preparing_workspace/roles.tsx:73]: nit: `''` is evaluated as falsy

File: webapp/channels/src/components/preparing_workspace/roles.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+const Roles = ({role, next, ...props}: Props) => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const categories = useSelector(getWorkTemplateCategories);
+
+    let className = 'Roles-body';
+
+    useEffect(() => {
+        dispatch(fetchCategories());
+    }, [dispatch]);
```
Comment (julmondragon) [inline webapp/channels/src/components/preparing_workspace/roles.tsx:40]: nit: Guessing that you want categories to be fetched once after mount, maybe adding `dispatch` as dep could cause it to be fetched multiple times, and causing re-renders.

**Review by julmondragon — COMMENTED** [review]
Nice work @JulienTant ! Just some nits and comments:

- Not sure if mobile view is being taken into account for the onboarding process, but it looks like the illustrations are covering some of the steps. (FWIW In other pages these are hidden for mobile - ie. login, signup): 

Screen Shot 2023-04-26 at 9 27 49 AM
Screen Shot 2023-04-26 at 9 29 15 AM

- The input seems to have a bit of the background illustration showing inside it, maybe need to add a background color to the container:
Screen Shot 2023-04-26 at 9 28 16 AM

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant :shipit:

**Review by julmondragon — APPROVED** [review]
Thanks @JulienTant !

File: webapp/channels/src/i18n/en.json [hunk 4]
```diff
@@ -4231,6 +4231,9 @@
   "onboarding_wizard.plugins.zoom": "Zoom",
   "onboarding_wizard.plugins.zoom.tooltip": "Start Zoom audio and video conferencing calls in Mattermost with a single click",
   "onboarding_wizard.previous": "Previous",
+  "onboarding_wizard.roles.description": "We’ll use this to suggest templates to help get you started.",
+  "onboarding_wizard.roles.other_input_placeholder": "Please share your primary function.",
```
[the remark this answers is not in the document]
Comment (JulienTant) [reply webapp/channels/src/i18n/en.json:4235]: stranger that the CI did not catch this

**Comment by JulienTant** [comment]
up @annelieseklein @stevemudie !

**Review by stevemudie — APPROVED** [review]
Looks good @JulienTant!

**Pull request merged**